client/OPUNetGameSelectWnd.cpp

```diff
@@ -347,62 +347,17 @@ void OPUNetGameSelectWnd::OnTimer()
 	searchTickCount++;
 	if (searchTickCount >= SearchTickInterval)
 	{
-		searchTickCount = 0;
-
-		// First check a game server for games (if info is setup)
-		char addrString[MaxServerAddressLength];
-		opuNetTransportLayer->GetGameServerAddressString(addrString, sizeof(addrString));
-		if (addrString[0] != 0)
-		{
-			// Check the game server for a list of games
-			opuNetTransportLayer->SearchForGames(addrString, DefaultGameServerPort);
-		}
-		else
-		{
-			// Game server not available. Broadcast a search query  (Broadcast to LAN)
-			opuNetTransportLayer->SearchForGames(nullptr, config.GetInt(sectionName, "ClientPort", DefaultClientPort));
-		}
+		SearchForGames();
 	}
 
 	if ((joinAttempt > 0) && (joiningGame != nullptr))
 	{
-		joinAttemptTickCount++;
-		if (joinAttemptTickCount >= JoinAttemptInterval)
-		{
-			joinAttemptTickCount = 0;
-
-			if (joinAttempt > MaxJoinAttempt)
-			{
-				joinAttempt = 0;
-				joiningGame = nullptr;
-
-				SetStatusText("Game join failed");
-			}
-			else
-			{
-				joinAttempt++;
-				// Resend the Join request
-				opuNetTransportLayer->JoinGame(*joiningGame, joinRequestPassword);
-			}
-		}
+		UpdateJoinAttempt();
```

Comment:
You could reasonably call this one `RequestJoinGame`.

In an ideal world, the client would make a single request and be done with it. However, because of the possibility of dropped packets, it may send several attempts before giving up. The timer is to handle re-send of those join attempts, which it only does if there was no reply within some short interval. In that case, it assumes either the join request packet was dropped, and the host never got the request, or the reply was dropped, and the host knows about the client, but the client doesn't know the host knows.

Reply:
I wanted to indicate in the function name that this is not the initial join request, but updating the process of joining.

Reply:
I suppose I didn't check what functions already existed. Not sure how similar this code is to other existing code.

Maybe `RequestJoinGameRetry`, if the other function name already exists?